uropa/visualization.py

```diff
@@ -1,77 +1,125 @@
-# TODO imports
+import pandas as pd
+import seaborn as sns
+import matplotlib.pyplot as plt
 
 # -------------------- plot functions -------------------- #
 
 
-def distribution_plot(table, var, kind):
+def distribution_plot(table, var, kind="histogram", title=None, output=None, dpi=300):
     """
     Plot distribution of the selected numerical variable.
     Distribution can be shown as boxplot, violinplot or histogram/kde.
 
     Parameters
     ----------
     table : pd.DataFrame
-        Pandas dataframe containing the data.
-    var : <datatype>, <default value>
-        <param description>
-    kind : <datatype>, <default value>
-        <param description>
-    TODO add more parameters
+        Pandas dataframe containing the data
+    var : string
+        Column to be displayed
+    kind : string, default "histogram"
+        Kind of plot: "histogram", "boxplot" or "violin"
+    title : string, default None
+        Title of the plot
+    output : string, default None
+        Path where the plot should be saved
+    dpi : Int, default 300
+        Resolution of the plot
 
     Returns
     -------
-    <datatype> :
-        <return description>
-    TODO should return the plotting object
+    matplotlib.axes._subplots.AxesSubplot :
+        Plot object for further processing
     """
-    pass
 
+    sns.set_style("darkgrid")
 
-def count_plot(table, var, kind):
-    """
-    Count and plot the occurence of the selected categorical variable.
-    Either shown as a pie chart or bar plot.
+    match kind:
+        case "histogram":
+            distPlot = sns.histplot(data=table[var])
+        case "boxplot":
+            distPlot = sns.boxplot(y=table[var])
+        case "violin":
+            distPlot = sns.violinplot(x=table[var])
+        case _:
+            raise Exception(
+                f"{kind} not supported. Consider using one of the supported plots (histogram, boxplot or violin).")
 
-    Parameters
-    ----------
-    table : pd.DataFrame
-        Pandas dataframe containing the data.
-    var : <datatype>, <default value>
-        <param description>
-    kind : <datatype>, <default value>
-        <param description>
-    TODO add more parameters
+    if title:
+        distPlot.set(title=title)
 
-    Returns
-    -------
-    <datatype> :
-        <return description>
-    TODO should return the plotting object
-    """
-    pass
+    if output:
+        plt.savefig(output, dpi=dpi)
+
+    return distPlot
 
 
-def peak_count_plot(table, var, kind):
+def peak_count_plot(table, var="feature", peak_type="exon", group_by=["peak_chr", "peak_start", "peak_end", "peak_strand"], stacked=False, color_by=None, title=None, output=None, dpi=300):

```

Comment:
check parameters `var, feature, peak_type, group_by, color_by`. Also, I can't select `nan` with `peak_type` as `nan` can only be checked with `.isna()`.